mapie/classification.py

```diff
@@ -943,15 +952,38 @@ def fit(
 
         estimator = check_estimator_classification(
             X,
-            y_enc,
+            y,
             cv,
             self.estimator
         )
         self.n_features_in_ = check_n_features_in(X, cv, estimator)
 
         n_samples = _num_samples(y)
-        self.n_classes_ = len(np.unique(y))
-        self.classes_ = np.unique(y)
+
+        if self.cv == "prefit":
+            self.classes_ = estimator.classes_
+            self.n_classes_ = len(np.unique(self.classes_))
+            if self.n_classes_ == len(np.unique(y)):
+                pass
+            elif self.n_classes_ > len(np.unique(y)):
+                warnings.warn(
+                    "WARNING: your calibration dataset has less labels"
+                    + " than your training dataset (training"
+                    + f" has {self.n_classes_} unique labels while"
+                    + f" calibration have {len(np.unique(y))} unique labels"
+                )
+            else:
+                raise ValueError(
+                    "You have more labels in the y of your calibration dataset"
+                    + " than during training (training"
+                    + f" has {len(np.unique(estimator.classes_))} unique"
+                    + f" labels while calibration have {len(np.unique(y))}"
+                    + " unique labels"
+                )
+
+        else:
+            self.n_classes_ = len(np.unique(y))
```

Comment:
I think it would make more sense to have these variables be set once we have trained the estimator (in the no-prefit context)? And then have a test later to check the coherence of the results?

Reply:
done